internal/plugin/server.go

```diff
@@ -269,13 +275,19 @@ func (plugin *nvidiaDevicePlugin) ListAndWatch(e *pluginapi.Empty, s pluginapi.D
 		return err
 	}
 
+	// If health provider not available, wait for context cancellation
+	if plugin.healthProvider == nil {
+		<-plugin.ctx.Done()
+		return nil
+	}
```

Comment:
Under which conditions is the healthProvider `nil`? Could we not rather ALWAYS use at least a "no-op" healthProvider to ensure that we don't need to special case this here or at any point where we call `Start` or `Stop`?